src/main/java/com/google/devtools/build/lib/actions/cache/VirtualActionInput.java

```diff
@@ -67,6 +67,11 @@ public byte[] atomicallyWriteRelativeTo(Path execRoot, String uniqueSuffix) thro
   protected byte[] atomicallyWriteTo(Path outputPath, String uniqueSuffix) throws IOException {
     Path tmpPath = outputPath.getFileSystem().getPath(outputPath.getPathString() + uniqueSuffix);
```

Comment:
The fact that you're seeing this problem (and we've seen some similar ones) just goes to show that the `uniqueSuffix` is insufficiently unique. We should rather add something truly unique here and have the `uniqueSuffix` just serve to indicate what it's used for (which is most of the usage I see today).

Reply:
Actually using a deterministic suffix here is actually not that bad. It means that if the process is repeated, any stale temporary files get cleaned up automatically, right?